conan/tools/env/environment.py

```diff
@@ -534,18 +574,27 @@ def save_script(self, filename):
         if ext:
             is_bat = ext == ".bat"
             is_ps1 = ext == ".ps1"
+            is_fish = ext == ".fish"
         else:  # Need to deduce it automatically
             is_bat = self._subsystem == WINDOWS
             is_ps1 = self._conanfile.conf.get("tools.env.virtualenv:powershell", check_type=bool)
-            if is_ps1:
+            is_fish = self._conanfile.conf.get("tools.env.virtualenv:fish", check_type=bool)
```

Comment:
There are no wrappers for ``.fish`` files, are they? If there are no wrappers, this will break when set and used to build any dependency. 
This should most likely apply only to end-consumer conanfile, not cache dependencies, and only if not used for ``conan build`` or anything like that.

Reply:
Done, please, take a look in the test `test_transitive_tool_requires`. It generates a tool requirement, then a second recipe consumes it and runs an executable that prints an environment variable. All using Fish.